From d0e6d2b80cc1461cc212951ebabaa8e44542c649 Mon Sep 17 00:00:00 2001
Message-ID: <d0e6d2b80cc1461cc212951ebabaa8e44542c649.1731584861.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1731584860.git.sharlatanus@gmail.com>
References: <cover.1731584860.git.sharlatanus@gmail.com>
From: Ricardo Wurmus <rekado@elephly.net>
Date: Thu, 14 Nov 2024 11:45:38 +0000
Subject: [PATCH v2 14/14] gnu: Add awscli-2.

* gnu/packages/python-web.scm (awscli-2): Rename to...
(awscliv2): ...this new variable.
(awscli-2): New variable.

Change-Id: Id5c356ae0e6af53167d4acde7934f88c2f527a00
---
 gnu/packages/python-web.scm | 71 ++++++++++++++++++++++++++++++++++++-
 1 file changed, 70 insertions(+), 1 deletion(-)

diff --git a/gnu/packages/python-web.scm b/gnu/packages/python-web.scm
index d3ef984a07..bc18c282c6 100644
--- a/gnu/packages/python-web.scm
+++ b/gnu/packages/python-web.scm
@@ -4285,9 +4285,78 @@ (define-public awscli
 Services (AWS) API.")
     (license license:asl2.0)))
 
+(define-public awscli-2
+  (package
+    (inherit awscli)
+    (name "awscli")
+    (version "2.20.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/aws/aws-cli")
+             (commit version)))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32
+         "0hyr9gmcfk7nzkgs0v6wgkh8k15dyhknqzfymbc9a9sa2dblc40q"))))
+    (build-system pyproject-build-system)
+    (arguments
+     (list
+      #:phases
+      #~(modify-phases %standard-phases
+          (add-after 'unpack 'ignore-deprecations
+            (lambda _
+              (substitute* "pyproject.toml"
+                (("\"error::") "\"ignore::"))))
+          (add-after 'unpack 'remove-pep517
+            (lambda _
+              (rename-file "backends/pep517.py" "backends/dummypep517.py")
+              (substitute* "pyproject.toml"
+                (("pep517") "dummypep517"))
+              (setenv "PYTHONPATH"
+                      (string-append (getcwd) ":"
+                                     (getcwd) "/backends:"
+                                     (getenv "GUIX_PYTHONPATH")))))
+          (add-after 'unpack 'fix-reference-to-groff
+            (lambda* (#:key inputs #:allow-other-keys)
+              (substitute* "awscli/help.py"
+                (("if self._exists_on_path\\('groff'\\):") "if True:")
+                (("cmdline = \\['groff'")
+                 (string-append "cmdline = ['"
+                                (search-input-file inputs "bin/groff")
+                                "'")))))
+          (replace 'check
+            (lambda* (#:key tests? #:allow-other-keys)
+              (when tests?
+                (substitute* "scripts/ci/run-tests"
+                  (("--numprocesses=auto --dist=loadfile --maxprocesses=4") ""))
+                ;; For an unknown reason pytest receives SIGTERM and no tests
+                ;; are run..
+                #;
+                (invoke "python" "scripts/ci/run-tests")))))))
+    (inputs
+     (list groff
+           python-awscrt-for-awscli
+           python-colorama
+           python-botocore
+           python-cryptography
+           python-dateutil
+           python-docutils
+           python-jmespath
+           python-prompt-toolkit
+           python-ruamel.yaml-0.16
+           python-ruamel.yaml.clib
+           python-urllib3))
+    (native-inputs
+     (list python-distro
+           python-flit
+           python-pytest
+           python-wheel))))
+
 ;; This is not an official release of awscli version 2, so it should not be
 ;; named awscli.
-(define-public awscli-2
+(define-public awscliv2
   (package
     (inherit awscli)
     (name "awscliv2")
-- 
2.46.0

